lib/matplotlib/axes/_axes.py

```diff
@@ -4263,18 +4260,17 @@ def _parse_scatter_color_args(c, edgecolors, kwargs, xshape, yshape,
             try:  # Then is 'c' acceptable as PathCollection facecolors?
                 colors = mcolors.to_rgba_array(c)
                 n_elem = colors.shape[0]
```

Comment:
then we define n_elem here, and don't use in the next line...

Reply:
Again, this is the way it was, and as far as I can see, it still works as intended, providing information for an error message.